cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -662,7 +663,11 @@ Status NumPyConverter::Visit(const StringType& type) {
   char numpy_byteorder = dtype_->byteorder;
 
   // For Python C API, -1 is little-endian, 1 is big-endian
+#if ARROW_LITTLE_ENDIAN
```

Comment:
`byteorder` has `|` (native) regarding built-in type. Thus, we use `ARROW_LITTLE_ENDIAN` to handle `|`.

Reply:
Could you add a comment here to this effect (that both `|` and `<` yield little-endian in the first case and `|` and `>` yield big-endian in the second case)?